src/Views/FirmwareView.vala

```diff
@@ -354,8 +354,8 @@ public class About.FirmwareView : Granite.SimpleSettingsPage {
 
     private void show_reboot_dialog () {
         var message_dialog = new Granite.MessageDialog.with_image_from_icon_name (
-            _("An update requires the system to restart to complete"),
```

Comment:
Double _to_.
To complete what? Left in the air.
The whole sentence is turned on its head, because the premise is that the system is required to restart, but only at the last words is it revealed to be "to complete". This again references "An update" all the way at the beginning.
It is taxing to read this kind of spaghetti-language, and it is made worse yet in translation.